lib/lf/intergridfe/mesh_function_transfer.h

```diff
@@ -0,0 +1,91 @@
+#ifndef LF_INTERGRIDFE_MESH_FUNCTION_TRANSFER_H
+#define LF_INTERGRIDFE_MESH_FUNCTION_TRANSFER_H
+
+#include <lf/refinement/mesh_hierarchy.h>
+#include <lf/uscalfe/uscalfe.h>
+#include <Eigen/Dense>
+
+namespace lf::intergridfe {
+
+/**
+ * @brief Lift a MeshFunctionFE to a finer mesh inside a MeshHierarchy
+ * @tparam SCALAR_FE_COARSE The scalar of the FESpace on the coarse mesh
+ * @tparam SCALAR_FE_FINE The scalar of the FESpace on the fine mesh
+ * @tparam SCALAR_COEFF The type of the basis function coefficients
+ * @param mh The MeshHierarchy on which to lift the MeshFunction
+ * @param fes_fine The FESpace on the fine mesh
+ * @param mf The MeshFunctionFE to lift to a finer mesh
+ */
+template <typename SCALAR_FE_COARSE, typename SCALAR_FE_FINE,
+          typename SCALAR_COEFF>
+[[nodiscard]] Eigen::Matrix<SCALAR_FE_FINE, Eigen::Dynamic, 1>
+transferCoarseFineFE(
+    const lf::refinement::MeshHierarchy& mh,
+    const lf::uscalfe::UniformScalarFESpace<SCALAR_FE_FINE>& fes_fine,
+    const lf::uscalfe::MeshFunctionFE<SCALAR_FE_COARSE, SCALAR_COEFF>& mf) {
+  // Find the index of the coarse mesh in mh
+  const lf::base::size_type mesh_index = [&]() {
+    lf::base::size_type idx = 0;
+    for (const auto mesh : mh.getMeshes()) {
+      if (mesh == mf.getMesh()) {
+        return idx;
+      }
+      ++idx;
+    }
+    return idx;
+  }();
+  // Assert whether the user provided a valid mesh
+  LF_ASSERT_MSG(mesh_index < mh.NumLevels() - 1,
+                "Invalid Mesh provided to function.");
+  // Get the refinement info of the meshes
+  const auto& child_info = mh.CellChildInfos(mesh_index);
+  // Iterate over all coarse cells and lift the coefficients
+  const auto mesh_fine = fes_fine.Mesh();
+  const auto& dofh_fine = fes_fine.LocGlobMap();
+  const lf::uscalfe::FeSpaceLagrangeO1<SCALAR_FE_FINE> fes_lagr_o1_fine(
+      mesh_fine);  // Used for coordinate transformations
+  Eigen::Matrix<SCALAR_FE_FINE, Eigen::Dynamic, 1> coeffs_fine(
+      dofh_fine.NumDofs());
+  const auto cells = mf.getMesh()->Entities(0);
+  for (size_t cell_idx = 0; cell_idx < cells.size(); ++cell_idx) {
+    const auto& cell = *cells[cell_idx];
+    // Reconstruct the refinement pattern
+    const lf::refinement::Hybrid2DRefinementPattern refpat(
+        cell.RefEl(), child_info[cell_idx].ref_pat_,
+        child_info[cell_idx].anchor_);
```

Comment:
I support the idea to offer a `GeometryInParent()` functionality. This would have to be implemented in `lf::refinement`, probably as a member function of `MeshHierarchy`.

Reply:
`GeometryInParent()` is available now.